src/core/ext/xds/xds_channel_creds.cc

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright 2019 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "absl/container/flat_hash_map.h"
+
+#include "src/core/ext/xds/xds_channel_creds.h"
+#include "src/core/lib/security/credentials/fake/fake_credentials.h"
+
+namespace grpc_core {
+
+namespace {
+
+using ChannelCredsMap =
+    absl::flat_hash_map<absl::string_view,
+                        std::unique_ptr<XdsChannelCredsImpl>>;
+ChannelCredsMap* g_creds = nullptr;
+
+
+}  // namespace
+
+//
+// XdsChannelCredsImpl implementations for default-supported cred types.
+//
+
+class GoogleDefaultXdsChannelCredsImpl : public XdsChannelCredsImpl {
+ public:
+  RefCountedPtr<grpc_channel_credentials> CreateXdsChannelCreds(
+      const Json& /*config*/) const override {
+    return RefCountedPtr<grpc_channel_credentials>(
+        grpc_google_default_credentials_create(nullptr));
+  }
+  bool IsValidConfig(const Json& /*config*/) const override { return true; }
+  absl::string_view creds_type() const override { return "google_default"; }
```

Comment:
Please define this method first in each of these classes.

Reply:
Sorry that I missed this, fixed.